images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,339 @@
+    required: true
+    type: string
+  involvedObject:
+    description: ObjectReference
+      - Object event is reporting on. ApiVersion, kind, name and namespace are of the involvedObject.
+    - apiVersion
+      required: true
+      type: string
+    - kind
+      required: true
+      type: string
+    - name
+      required: true
+      type: string
+    - namespace
+      required: true
+      type: string
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+"""
+
+EXAMPLES = """
+
+- name: Create Kubernetes Event
+  k8s_events
+    state: present
+    name: test-https-emily109
+    namespace: default
+    message: Event created
+    reason: Created
+    reportingComponent: Reporting components
+    type: Normal
+    source:
+    component: Metering components
+    involvedObject:
+      apiVersion: v1
+      kind: Service
+      name: test-https-emily107
+      namespace: default
+"""
+
+RETURN = """
+result:
+  description:
+  - The created, patched, or otherwise present Event object. Will be empty in the case of a deletion.
+  returned: success
+  type: complex
+  contains:
+     api_version:
+        description: The versioned schema of this representation of an object.
+        returned: success
+        type: str
+     count:
+       description: Count of Event occurance
+       returned: success
+       type: int
+     firstTimestamp:
+       description: Timestamp of first occurance of Event
+       returned: success
+       type: timestamp
+     involvedObject:
+       description: Object event is reporting on. Includes apiVersion, kind, name, namespace, resourceVersion and uid.
+       returned: success
+       type: complex
+     kind:
+       description: Always 'Event'.
+       returned: success
+       type: str
+     lastTimestamp:
+       description: Timestamp of last occurance of Event
+       returned: success
+       type: timestamp
+     message:
+       description: Status for operation
+       returned: success
+       type: str
+     metadata:
+       description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+       returned: success
+       type: complex
+     reason:
+       description: Reason for the transition into the objects current status
+       returned: success
+       type: str
+     reportingComponent:
+       description: Component responsible for event
+       returned: success
+       type: str
+     source:
+       description: Component for reporting this Event
+       returned: success
+       type: string
+     type:
+       description: Type of Event. Either Normal or Warning
+       returned: success
+       type: string
+
+"""
+
+import copy
+import traceback
+import datetime
+import kubernetes.config.dateutil
+import openshift
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+# user args
+EVENT_ARG_SPEC = {
```

Comment:
I think we should remove this comment too, as this pattern of defining a dictionary of arguments is pretty common in modules.